module/util/util.go

```diff
@@ -145,3 +145,42 @@ func WaitError(errChan <-chan error, done <-chan struct{}) error {
 		return nil
 	}
 }
+
+// UnboundedChannel returns a pair of channels that represent the two ends of
```

Comment:
I'm wary of making use of unbounded channels in general. For the specific case they were used in this PR, I suggested an alternative [here](https://github.com/onflow/flow-go/pull/2841#discussion_r939451895). 

In general, I think we should avoid using unbounded channels because:
* they can hide performance problems and bugs, especially when the underlying buffer size isn't visible in any way at runtime
* there are usually better options to manage the back-pressure:
  * we can filter/drop the incoming messages (in the use case here, we only ever care about the largest height value in the queue, everything else can be dropped)
  * we can block the message producer

👉  I'm personally inclined to remove this function altogether, especially if we have consensus on the `MonotonousCounter` suggestion to replace the use-case in this PR. Then `UnboundedChannel` would be unused.